mail_tracking/models/ir_mail_server.py

```diff
@@ -80,9 +80,11 @@ def send_email(self, message, mail_server_id=None, smtp_server=None,
         except Exception as e:
             if tracking_email:
                 tracking_email.smtp_error(self, smtp_server_used, e)
-        if message_id and tracking_email:
-            vals = tracking_email._tracking_sent_prepare(
-                self, smtp_server_used, message, message_id)
-            if vals:
-                self.env['mail.tracking.event'].sudo().create(vals)
+            raise
+        finally:
+            if message_id and tracking_email:
+                vals = tracking_email._tracking_sent_prepare(
+                    self, smtp_server_used, message, message_id)
+                if vals:
+                    self.env['mail.tracking.event'].sudo().create(vals)
```

Comment:
```suggestion
            raise
        if message_id and tracking_email:
            vals = tracking_email._tracking_sent_prepare(
                self, smtp_server_used, message, message_id)
            if vals:
                self.env['mail.tracking.event'].sudo().create(vals)
```

Reply:
That's to say, just add the `raise` line.

Reply:
Yes, but why? What's the benefit?

Reply:
Several:

- Less diff (only one line)
- Don't create something that is going to be rolled-back.
- I even doubt if this will generate other exception in that state.

Reply:
> Don't create something that is going to be rolled-back.

But it is not going to be rolled back... :roll_eyes:

Reply:
That happens automatically when an exception is raised.

Reply:
Yes, every time this method is called, it is done inside a try-except block that avoids the rollback.

Reply:
OK